Add authorship policy (#726)
GenericMappingTools/pygmt: AUTHORSHIP.md

Every time we make a release, everyone who has made a commit to the repository
since the previous release will be mentioned in the changelog entry. If their
full name is available on GitHub, we will use it. Otherwise, we will use the
GitHub handle. This is a way of saying "Thank you".

## Authorship on Zenodo archives of releases

Anyone who has contributed to the repository (i.e., appears on `git log`) will
be invited to be an author on the Zenodo archive of new releases.

To be included as an author, you *must* add the following to the `AUTHORS.md`
file of the repository:

1. Full name
2. Affiliation (if omitted, we will use "Unaffiliated")
3. ORCID (optional)

The order of authors will be defined by the number of commits to the repository
(`git shortlog -sne`). The order can also be changed on a case-by-case basis.

If you have contributed and do not wish to be included in Zenodo archives,
there are a few options:

1. Don't add yourself to `AUTHORS.md`
2. Remove yourself from `AUTHORS.md`
3. Indicate next to your name on `AUTHORS.md` that you do not wish to be
   included with something like `(not included in Zenodo)`.

## Scientific publications (papers)

> We aim to write academic papers for most of your software packages. Ideally, we
> will publish updated papers for major changes or large new components of the

Reviewer: ```suggestion
We aim to write academic papers for most of our software packages. Ideally, we
will publish updated papers for major changes or significant new components of the
```
Possible typo? Might want to think this section through a bit.

Reviewer 2: @weiji14  My apologies; I didn't think to formally suggest a change and instead just commented. Thanks for doing it!

Reviewer: No worries, and do feel free to review stuff. I don't want us 'maintainers' to come across as the only people making big decisions. This authorship policy might come across as benefiting more established people, but a line does need to be drawn somewhere, and we want to make sure that we get the balance right.

package.

To be included as an author on the paper, you *must* satisfy the following
criteria:

1. Have made a contribution to the repository or significant non-coding

>    contributions.

Reviewer: Might be good to spell out what *significant* means here. Any good ideas?

As for defining what *contribution* means, would following https://casrai.org/credit/ be a good idea?

Reviewer 2: @weiji14 My apologies; I didn't think to formally suggest a change and instead just commented. Thanks for doing it!

Reviewer 2: ```suggestion
1. Have made multiple and regular contributions to the repository, or the GMT repository, in numerous facets, to include wrapping functions, testing, and writing documentation.
2. Have made non-coding contributions, including project administration and decision making.
3. Have participated in the writing and reviewing of the paper.
```

This is just a rough attempt at clarifying the authorship responsibilities and expectations, but as a disclaimer I have also never been an author on a paper. My goal is to not be too specific, as it's not like there should be a magic number of functions wrapped/tests written/lines of code contributed/Zoom meetings attended to be listed as an author. But I think this spells out that the intent that authors must be individuals who are involved in the different aspects of the project, and the expectation is that participation extends beyond contributing code.

Reviewer 3: I think this is a good start to a definition of 'significant contributions'.  I interpret the current statement "to include wrapping functions, testing, and writing documentation" to mean that all three aspects are required for co-authorship and would prefer the language "such as wrapping functions, testing, and/or writing documentation" because I think that PyGMT will benefit from *not* mandating coding contributions (versus strictly documentation improvements) for recognition through co-authorship.

Reviewer 2: I think that's a good change @meghanrjones

Reviewer 4: Totally agree @meghanrjones.

Author: Are we going to commit the suggested changes in https://github.com/GenericMappingTools/pygmt/pull/726#discussion_r562821369 and update it based on feedbacks from https://github.com/GenericMappingTools/pygmt/pull/726#discussion_r573026686?

Reviewer 2: I updated it to Meghan's version and committed this change in 021983a
